src/CLR/Core/TypeSystem.cpp

```
@@ -9114,29 +9114,34 @@ static bool MatchSignatureForVirtualDispatch(CLR_RT_SignatureParser &parserLeft,
         // Relaxed VAR <-> GENERICINST matching for interface virtual dispatch:
         // the interface method may use a type parameter (VAR N) while the concrete
         // implementation uses the expanded generic type (GENERICINST ...).  Drain the
-        // GENERICINST inner elements so the parser position stays consistent.
+        // whole GENERICINST sub-tree on the expanded side so the parser position stays
+        // consistent.
         if (resLeft.DataType == DATATYPE_VAR && resRight.DataType == DATATYPE_GENERICINST)
```

Clang (14.0.6)

[note] +2, including nesting penalty of 1, nesting level increased to 2

(clang)